roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/StreetSection.java

```diff
@@ -55,7 +57,11 @@ public void updateAllCarsPositions() {
     }
 
     public boolean isFirstCarOnExitPoint() {
-        return false;
+       ICar firstCar = this.getFirstCar();
+        if(firstCar == null){
+            return false;
+        }
+        return DoubleMath.fuzzyEquals(this.getCarPositions().get(firstCar), this.getLength(), 0.5); // Todo: sergii firstCar.getDriverBehaviour().getMaxDistance()
```

Comment:
split it, to be more readable
carPositions = this.getCarPositions();
firstCarPosition = carPositions.get(firstCar);